src/Internal/Wallet.purs

```diff
@@ -60,77 +77,138 @@ data Wallet
   | Lode Cip30Wallet
   | KeyWallet KeyWallet
 
+data WalletExtension
```

Comment:
I like how removing `KeyWallet` from this type simplified the code and at the same time, it forbade attempts to call functions that make nonsense for `KeyWallet` (like `icon`). Nice!